### ISSUE TITLE
Some CSS is missing with dev preview

### ISSUE THREAD

**Issue by bojanrajh** [unocss/unocss#2205]
### UnoCSS version

\>=0.49.0

### Describe the bug

Some CSS is missing when using the `dev` preview. It is correctly built using the `build` command in VitePress.

Missing CSS:
```css
*, ::before, ::after {
    // ...
    --un-ring-color: rgba(147,197,253,0.5);
}
```

| version | dev | build |
|---|---|---|
|>=49.0|**broken**|OK|
|<=48.5|OK|OK|

### Reproduction

See [StackBlitz > Example](https://stackblitz.com/edit/vite-nwgpgr?file=docs/example.md).

### System Info

/

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/unocss/unocss/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't [already an issue](https://github.com/unocss/unocss/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/unocss/unocss/discussions) or join our [Discord Chat Server](https://chat.antfu.me).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sibbng** [comment]
If we are looking for `--un-ring-color` variable it appears on me. I can't see any issue on repro. Can you give more details about the issue?

**Comment by bojanrajh** [comment]
\>= v49.0
![image](https://user-images.githubusercontent.com/117360292/220100325-1963d891-029b-49c8-8d1a-1541b3d39700.png)

<= v48.5
![image](https://user-images.githubusercontent.com/117360292/220100389-a21fa081-297d-405e-bf3f-20fd4b93d4ed.png)

This CSS is missing on >= v49.0:
![image](https://user-images.githubusercontent.com/117360292/220100865-f1f603f7-7a5e-4e3d-b541-b1e339f2d003.png)

The same CSS section is added to `::backdrop` in all versions.
![](https://media2.giphy.com/media/v1.Y2lkPTc5MGI3NjExOTAyYjQwNWRhNmViNzA4OTA2YTA2ZGEwYzBmNDRkNDNmOGE2YmE2OSZjdD1n/5XRAlzFyeVjXJ9n2pH/giphy.gif)

**Comment by sibbng** [comment]
I upgraded unocss to ^0.49.0 on your repro, and the shadow still appears.

![image](https://user-images.githubusercontent.com/19991745/220112507-03ebbc54-dceb-4040-8cb7-96f7e55a16cf.png)

**Comment by bojanrajh** [comment]
Seems it was fixed in v49.7 (probably [here](https://github.com/unocss/unocss/pull/2204).

![image](https://user-images.githubusercontent.com/117360292/220119760-c5f04b8f-e625-49c5-8eed-03f44f23d963.png)

Thanks!
